api/all/src/main/java/io/opentelemetry/api/trace/Span.kt

```diff
@@ -0,0 +1,499 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package io.opentelemetry.api.trace
+
+import io.opentelemetry.api.common.AttributeKey
+import io.opentelemetry.api.common.AttributeKey.Companion.booleanKey
+import io.opentelemetry.api.common.AttributeKey.Companion.doubleKey
+import io.opentelemetry.api.common.AttributeKey.Companion.longKey
+import io.opentelemetry.api.common.AttributeKey.Companion.stringKey
+import io.opentelemetry.api.common.Attributes
+import io.opentelemetry.api.common.Attributes.Companion.empty
+import io.opentelemetry.api.internal.ApiUsageLogger
+import io.opentelemetry.context.Context
+import io.opentelemetry.context.ImplicitContextKeyed
+import java.time.Instant
+import java.util.concurrent.TimeUnit
+import java.util.function.BiConsumer
+import javax.annotation.concurrent.ThreadSafe
+
+/**
+ * An interface that represents a span. It has an associated [SpanContext].
+ *
+ *
+ * Spans are created by the [SpanBuilder.startSpan] method.
+ *
+ *
+ * `Span` **must** be ended by calling [.end].
+ */
+@ThreadSafe
+interface Span : ImplicitContextKeyed {
```

Comment:
Consider implementing explicit context first as this is required, then assess how to implement implicit context?